dbt/adapters/redshift/connections.py

```diff
@@ -84,14 +108,18 @@ def __init__(self, credentials):
     def get_connect_method(self):
         method = self.credentials.method
         kwargs = {
-            "host": self.credentials.host,
+            "host": "",
+            "region": self.credentials.region,
             "database": self.credentials.database,
             "port": self.credentials.port if self.credentials.port else 5439,
             "auto_create": self.credentials.autocreate,
             "db_groups": self.credentials.db_groups,
-            "region": self.credentials.host.split(".")[2],
             "timeout": self.credentials.connect_timeout,
+            "application_name": "dbt",
         }
+        if method == RedshiftConnectionMethod.IAM or method == RedshiftConnectionMethod.DATABASE:
+            kwargs["host"] = self.credentials.host
+            kwargs["region"] = self.credentials.host.split(".")[2]
```

Comment:
if the user passes a region that's different from the region in the host should we raise an exception?

Reply:
Good catch, we could also default to using region from host, in the scenario that host is provided. So if provided region and region in host are different we use the region from host. But we could also throw an error. What are your thoughts?